src/gui/addnewtorrentdialog.cpp

```diff
@@ -679,6 +679,9 @@ void AddNewTorrentDialog::updateMetadata(const BitTorrent::TorrentInfo &metadata
     // Update UI
     setupTreeview();
     setMetadataProgressIndicator(false, tr("Metadata retrieval complete"));
+
+    if (!m_torrentInfo.infoHash().v2().isValid())
```

Comment:
Why the button visibility depends on v2 hash validness?

Reply:
We cannot produce valid .torrent file for v2 torrent at this point since we have no data for mandatory `piece layers` field.
Ref: https://github.com/arvidn/libtorrent/issues/6283

Reply:
Is https://github.com/arvidn/libtorrent/issues/6283#issuecomment-869742233 referring the same thing/fix for this? If so, then maybe we should wait a bit for it?

Reply:
>maybe we should wait a bit for it?

I think we shouldn't wait.
IIRC, requirement for `piece layers` is part of BitTorrent v2 specifications. I can't clearly imagine what fix can suggest @arvidn except the fix for BitTorrent specs.

Reply:
If that is the case, then I think the button should be grayed out/disabled instead of hiding it and maybe put ​tooltip stating that it is unable to work for v2 torrents (due to missing xxx data).

Reply:
> I think the button should be grayed out/disabled instead of hiding

We don't hide a button. There is no button shown until metadata is downloaded. Thus we only show button when we are able to create valid .torrent file.
Otherwise, won't it look weird if the button appears in the immutable disabled state?

Reply:
>We don't hide a button. There is no button shown until metadata is downloaded. Thus we only show button when we are able to create valid .torrent file.

The point is, for some casual users the button will sometimes be present and sometimes not and they won't ever know the cause of that and could suspect qbt is malfunctioning.

>Otherwise, won't it look weird if the button appears in the immutable disabled state?

IMO it signals that some obstruction is preventing the availability of this button/functionality.

Reply:
Is it OK with the latest push?